### DISCUSSION TITLE
Paused Free Plan projects are restorable for 90 days

### DISCUSSION THREAD

**Discussion by olirice** [supabase/supabase#27497] [Changelog]
_This only impacts projects on the Free Plan because projects in any of the paid plans cannot be paused._

Beginning June 24, 2024, we're updating some project pause/restore behavior:

- paused Free projects are restorable for 90 days following their pause date
- any Free projects paused before June 24 will be able to restore at any point before September 22, 2024 so they have a full 90 days from when this announcement is made
- once a project is no longer restorable, the "restore" option is replaced with an option to download the latest logical backup, taken right before the project is paused, and all Storage objects

This change is being made to enable us to maintain high development velocity on the platform. Previously, paused projects could be restored indefinitely. That creates the need for the platform to remain fully backwards compatible with outdated versions of Postgres and associated extensions. The update allows us to provide a reasonable pause/restore window while gaining the ability to evolve the platform.

[Some more relevant docs here](https://supabase.com/docs/guides/platform/upgrading)

**Comment by MyManAy** [comment]
Hi, I came here from the email announcement about the June 2024 Supabase changes. As a free tier user, I feel a little bit anxious and confused, and I was wondering if you could clarify a few details?

1. once the 90 days have passed since the pause date, do all my unrestored projects get wiped or lose data?
2. if free tier users can only have 2 projects active at a time, is it possible to restore more than 2?
3. is there anything free tier users can do to avoid losing data?

Thank you for your time.

**Reply by ProceduralPeasant** [reply]
" _once a project is no longer restorable, the "restore" option is replaced with an option to download the project's most recent backup_"

**Reply by MyManAy** [reply]
> " _once a project is no longer restorable, the "restore" option is replaced with an option to download the project's most recent backup_"

Does this mean that the project will automatically backup after it gets paused? If not, how are we able to manually backup our projects?

**Reply by darora** [reply]
A backup is taken as part of the pause process. At the end of the 90 day window, we no longer support restoring from it, but it can be downloaded from the dashboard. There is no risk of data loss.

You could still attempt to restore it manually to a new project (e.g. using `psql`), but might run into errors given changes in extensions etc, especially if a long period of time has passed.

**Reply by KelvinMatheka-1** [reply]
@darora darora the backup made by supabase during the pause dosn't seem to contain any of my recent data or functions. I was able to restre the backup downloaded frm the dashboard but nothing come close to the recent work I had, please advice

**Reply by DECODEDHQ** [reply]
hi< can anyone help me with the restore project situation? All my projects aren't working. It says it's a billing issue so I paid, yet they are still trying to charge me again while not restoring projects. I am confused, stressed and angry. Pls help me.  Supabase support team keep sending generic responses. How can I be on free plan after paying?

**Comment by sabatage3** [comment]
I have a question, what can i do with that backup that is available for downloading after the restorable period ends? I’m not self hosting so my question would be, can i restore that backup in a new supabase project?

**Reply by darora** [reply]
You can attempt to restore it using `psql` to a new Supabase project, or to a self-hosted environment, or really any other Postgres database.

As mentioned by @olirice , this change was made to allow the platform to move a little faster in terms of not having to maintain backwards compatibility with old versions of software. As such, the restore might run into issues or failures because of Postgres or extension upgrades/deprecations, and you might need to resolve these.

**Comment by Sergio-P** [comment]
What does this backup contain exactly? A copy of the DB tables, the storage, edge functions, logs?

**Reply by darora** [reply]
The backup is a full logical backup of the database, which can be restored to a Postgres cluster.

**Reply by Sergio-P** [reply]
So the storage is wiped out?

**Reply by w3b6x9** [reply]
You'll be able to download all of your project's Storage objects when the project has been paused for 90 days or longer. You can take this, along with your database's backup, and recreate in a new Supabase project.

**Reply by creix** [reply]
What about the edge functions? Is it possible to restore them? @darora

**Reply by w3b6x9** [reply]
@creix do you still have the edge functions saved in your local environment from when you wrote them?

**Reply by creix** [reply]
> @creix do you still have the edge functions saved in your local environment from when you wrote them?

In this specific case yes, I was just asking to know if in case was possible to restore them in some way

**Reply by w3b6x9** [reply]
@creix you can always create another Supabase project and deploy them there.

**Reply by creix** [reply]
@w3b6x9 Yeah, sorry I didn't explain myself well. I just wanted to ask if, in case I would not have them anymore, there would be a way to restore them from the project.

**Reply by KelvinMatheka-1** [reply]
@w3b6x9 w3b6x9, Just reestored the backup I downloaded to a new supabase project and it seems the back up is from march 2024 where as the project was checked and updated recently, meaning all the new work is gone? Kindly help.
![Screenshot 2025-02-19 130051](https://github.com/user-attachments/assets/cec29f04-a66f-4518-81d0-dd3e1a2930f6)

**Comment by chrissy0** [comment]
Would it be possible to keep the restore option enabled until a breaking change occurs? So a minimum of 90 days, but more if possible? I would assume that lots of paused projects will lose restore-ability for no reason other than an arbitrary 90 days having passed. I appreciate that this would introduce further complexity and might just not be worth it. But from a technical point of view, it should be possible.

**Comment by KTibow** [comment]
There seems to be a UI bug, I have to hit the restore button and get the error and reload before I can download the backup
